backend/generate.ts

```diff
@@ -0,0 +1,185 @@
+import Instructor from "@instructor-ai/instructor";
+import OpenAI from "openai";
+import { z } from "zod";
+import readline from "readline";
+
+const oai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY ?? undefined,
+});
+
+const client = Instructor({
+  client: oai,
+  mode: "TOOLS",
+});
+
+export const promptWithSchema = async (
+  prompt: string,
+  schema: z.ZodObject<any>
+) => {
+  try {
+    const response = await client.chat.completions.create({
+      messages: [{ role: "user", content: prompt }],
+      model: "gpt-3.5-turbo",
+      response_model: {
+        schema: schema,
+        name: "User",
+      },
+    });
+    return response;
+  } catch (error) {
+    console.error("Error generating prompt:", error);
+  }
+};
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const askQuestion = (question: string) => {
```

Comment:
Using an async function instead, it's not generally necessary to explicitly return a Promise.